lib/libp2p.ex

```diff
@@ -0,0 +1,149 @@
+defmodule Libp2p do
```

Comment:
I saw that rustler uses a separate `native` folder at the top to have the code that is not elixir. For example: https://github.com/rusterlium/NifIo

What do you think of following this approach?

Reply:
It can be seen in this blog post too: https://mainmatter.com/blog/2020/06/25/writing-rust-nifs-for-elixir-with-rustler/

Reply:
Done! I moved it so it uses a structure like this:

```
lib
|-- lambda_ethereum_consensus
|   |-- application.ex
|   `-- libp2p.ex
`-- lambda_ethereum_consensus.ex
native
`-- libp2p_nif
    |-- go.mod
    |-- go.sum
    |-- libp2p.c
    |-- main.a
    |-- main.go
    |-- main.h
    |-- utils.c
    `-- utils.h
priv
`-- native
    `-- libp2p_nif.so
```